File: state/java/InitializeState.java
package State;

// Farklı durumların meydana geldiği sınıf.
// UML diyagramındaki State (VendingMachineState) yapısından türer.
// UML diyagramındaki ConcreteState yapısına denk gelmektedir.
class InitializeState extends VendingMachineState {
    // Başlangıç durumu ayarları yapıldıktan sonra WaitingState (hazır) durumuna geçer.
    @Override
    public void handleState(VendingMachine context) {
        System.out.println("InitializeState: Ön hazırlıklar yapılıyor.");

        context.setState(new WaitingState());
    }
}

File: state/go/initialize-state.go
package main

import "fmt"

// initializeState Farklı durumların meydana geldiği sınıf.
// UML diyagramındaki State (VendingMachineState) yapısından türer.
// UML diyagramındaki ConcreteState yapısına denk gelmektedir.
type initializeState struct{}

// Başlangıç durumu ayarları yapıldıktan sonra WaitingState (hazır) durumuna geçer.

func NewInitializeState() VendingMachineState {
	return initializeState{}
}

func (s initializeState) HandleState(context *VendingMachine) {
	fmt.Println("initializeState: Ön hazırlıklar yapılıyor.")
	context.SetState(NewWaitingState())
}
